src/engine/AbsoluteValueConstraint.cpp

```diff
@@ -0,0 +1,454 @@
+        fLowerBound = 0;
+    }
+
+    if ( bLowerBound >= 0 )
+    {
+        // Positive phase, all bounds much match
+        tightenings.append( Tightening( _f, bLowerBound, Tightening::LB ) );
+        tightenings.append( Tightening( _b, fLowerBound, Tightening::LB ) );
+
+        tightenings.append( Tightening( _f, bUpperBound, Tightening::UB ) );
+        tightenings.append( Tightening( _b, fUpperBound, Tightening::UB ) );
+    }
+
+    else if ( bUpperBound <= 0 )
+    {
+        // Negative phase, all bounds must match
+
+        tightenings.append( Tightening( _f, -bUpperBound, Tightening::LB ) );
+        tightenings.append( Tightening( _b, -fUpperBound, Tightening::LB ) );
+
+        tightenings.append( Tightening( _f, -bLowerBound, Tightening::UB ) );
+        tightenings.append( Tightening( _b, -fLowerBound, Tightening::UB ) );
+    }
+
+    else if ( bLowerBound < 0 && bUpperBound >= 0 && FloatUtils::isZero( fLowerBound ) )
+    {
+        // Phase undetermined, b can be either positive or negative, f can be 0
+        tightenings.append( Tightening( _b, -fUpperBound , Tightening::LB ) );
+        tightenings.append( Tightening( _b, fUpperBound, Tightening::UB ) );
+        tightenings.append( Tightening( _f, FloatUtils::max( -bLowerBound , bUpperBound ), Tightening::UB ) );
+    }
+
+    else if ( bLowerBound < 0 && bUpperBound >= 0 && fLowerBound > 0 )
+    {
+        // Phase undetermined, b can be either positive or negative, f strictly positive
+        tightenings.append( Tightening( _b, -fUpperBound, Tightening::LB ) );
+        tightenings.append( Tightening( _b, fUpperBound, Tightening::UB ) );
+        tightenings.append( Tightening( _f, FloatUtils::max( -bLowerBound, bUpperBound ), Tightening::UB ) );
+
+        // Below we test if the phase has actually become fixed
+        if ( fLowerBound > -bLowerBound )
+        {
+            // Positive phsae
+            tightenings.append( Tightening( _b, fLowerBound, Tightening::LB ) );
+        }
+
+        if ( fLowerBound > bUpperBound )
+        {
+            // Negative phsae
+            tightenings.append( Tightening( _b, -fLowerBound, Tightening::UB ) );
+        }
+
+    }
+}
+
+void AbsoluteValueConstraint::getAuxiliaryEquations( List<Equation> &/* newEquations */ ) const
+{
+    // Currently unsupported
+}
+
+String AbsoluteValueConstraint::serializeToString() const
+{
+    // Output format is: Abs,f,b
+    return Stringf( "absoluteValue,%u,%u", _f, _b );
+}
+
+void AbsoluteValueConstraint::fixPhaseIfNeeded()
+{
+    // Option 1: b's range is strictly positive
+    if ( _lowerBounds.exists( _b ) && _lowerBounds[_b] >= 0 )
+    {
+        setPhaseStatus( PHASE_POSITIVE );
+        return;
+    }
+
+    // Option 2: b's range is strictly negative:
+    if ( _upperBounds.exists( _b ) && _upperBounds[_b] <= 0 )
+    {
+        setPhaseStatus( PHASE_NEGATIVE );
+        return;
+    }
+
+    if ( !_lowerBounds.exists( _f ) )
```

Comment:
What is this case?

Reply:
Initially, when the PL constraint is just created, it doesn't know any bounds yet - this is just to be on the safe side.